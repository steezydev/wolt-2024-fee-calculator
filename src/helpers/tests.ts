const localStorageMock = (function () {
    let store: Record<string, string> = {};
    return {
        getItem(key: string) {
            return store[key] || null;
        },
        setItem(key: string, value: string) {
            store[key] = value.toString();
        },
        clear() {
            store = {};
        },
    };
})();

const matchMediaMock = jest.fn().mockImplementation((query) => ({
    matches: query === '(prefers-color-scheme: dark)',
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
}));

export const mockContext = () => {
    beforeAll(() => {
        Object.defineProperty(window, 'localStorage', {
            value: {
                getItem: jest.fn().mockImplementation((key: string) => {
                    return localStorageMock.getItem(key);
                }),
                setItem: jest
                    .fn()
                    .mockImplementation((key: string, value: string) => {
                        localStorageMock.setItem(key, value);
                    }),
                clear: localStorageMock.clear,
            },
            writable: true,
        });

        window.matchMedia = matchMediaMock;
    });

    beforeEach(() => {
        (window.localStorage.getItem as jest.Mock).mockClear();
        (window.localStorage.setItem as jest.Mock).mockClear();
        matchMediaMock.mockClear();
    });
};

export const setupFakeDate = () => {
    beforeEach(() => {
        jest.useFakeTimers();
        jest.setSystemTime(new Date('2024-01-01'));
    });

    afterEach(() => {
        jest.useRealTimers();
    });
};

export const mockScrollTo = () => {
    beforeEach(() => {
        Element.prototype.scrollTo = jest.fn();
    });

    afterEach(() => {
        (Element.prototype.scrollTo as jest.Mock).mockRestore();
    });
};
