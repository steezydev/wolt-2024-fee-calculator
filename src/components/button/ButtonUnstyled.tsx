import { classNames } from '@/helpers/classNames';
import { ButtonProps } from '@/types/props/ButtonProps';

const ButtonUnstyled = ({
  id,
  children,
  ariaLabel,
  className,
  disabled,
  onClick,
  type,
  buttonRef,
}: ButtonProps) => {
  // Preventing form submission on button click
  const handleClick = (e: React.MouseEvent<HTMLButtonElement>) => {
    e.preventDefault();
    onClick && onClick();
  };

  return (
    <button
      ref={buttonRef}
      id={id}
      type={type}
      onClick={handleClick}
      aria-label={ariaLabel}
      className={classNames('', className)}
      disabled={disabled}
      data-test-id={id}
    >
      {children}
    </button>
  );
};

export default ButtonUnstyled;
