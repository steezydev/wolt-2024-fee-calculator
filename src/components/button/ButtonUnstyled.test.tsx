import { buttonEventTests, buttonRenderTests } from './Button.tests';
import ButtonUnstyled from './ButtonUnstyled';

buttonRenderTests('ButtonUnstyled', ButtonUnstyled);
buttonEventTests('ButtonUnstyled', ButtonUnstyled);
