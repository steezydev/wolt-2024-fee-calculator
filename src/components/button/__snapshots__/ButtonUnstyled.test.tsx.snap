// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonUnstyled when asked to render disabled matches snapshot 1`] = `
<button
  aria-label="Test button label"
  className=""
  data-test-id="test-button"
  disabled={true}
  id="test-button"
  type="button"
>
  Test Button
</button>
`;

exports[`ButtonUnstyled when asked to render with additional class matches snapshot 1`] = `
<button
  aria-label="Test button label"
  className="some-class"
  data-test-id="test-button"
  id="test-button"
  type="button"
>
  Test Button
</button>
`;

exports[`ButtonUnstyled when asked to render with children matches snapshot 1`] = `
<button
  aria-label="Test button label"
  className=""
  data-test-id="test-button"
  id="test-button"
  type="button"
>
  Test Button
</button>
`;
